lib/ReactViews/ExplorerWindow/Tabs/MyDataTab/AddData.jsx

```diff
@@ -103,19 +113,18 @@ const AddData = createReactClass({
       }
     }
     addUserCatalogMember(this.props.terria, promise).then(addedItem => {
-      if (addedItem && !(addedItem instanceof TerriaError)) {
-        this.onFileAddFinished(addedItem);
-      }
       this.setState({
         isLoading: false
       });
       this.props.resetTab();
+      if (addedItem && !(addedItem instanceof TerriaError)) {
+        this.onFileAddFinished([addedItem]);
```

Comment:
Can you explain this change?

Does moving this processing after setting `isLoading` to false mean that we don't have a loading state? I know the current synchronous loading of catalog items leads to UI freezes, but is is better this way?

Reply:
So when my custom component receives `onFileAddFinished` its state changes and the explorer component gets unmounted. If we call, `setState` after the unmount has happened, react throws an error (or is it a warning). But you are right, changing the order here is not correct because we don't get the loading indicator for the item load. I'll think of a fix.

Reply:
I have reverted the change for now.